debian/common_context/entrypoint.sh

```diff
@@ -9,27 +9,44 @@
 #   Felix Schindler (2017)
 #   Rene Milk       (2017)
 
-USERNAME_=${LOCAL_USER:-user}
-UID_=${LOCAL_UID:-1000}
+USERNAME_=${LOCAL_USER:-root}
+UID_=${LOCAL_UID:-0}
 GID_=${LOCAL_GID:-$UID_}
 
-groupadd -g $GID_ $USERNAME_ &> /dev/null
-if [ -e /home/$USERNAME_ ] ; then
-  useradd -d /home/$USERNAME_ -g $GID_ -s /bin/bash -u $UID_ $USERNAME_
-else
-  useradd -md /home/$USERNAME_ -g $GID_ -s /bin/bash -u $UID_ $USERNAME_
-fi
+export LANG=en_US.UTF-8
+echo "127.0.0.1 ${HOSTNAME}" >> /etc/hosts
 
-export ALLINEA_LICENCE_FILE=/home/$USERNAME_/.config/allinea/Licence \
+# we are running as root, if UID_ == 0, there is nothing we can do but continue ...
+if [[ $UID_ == 0 ]] ; then
+  if [ "X$@" == "X" ]; then
+    exec /bin/bash
+  else
+    exec "$@"
+  fi
+  exit
+fi
 
-chown -R $USERNAME_:$GID_ /home/$USERNAME_
+# ... else, check if there exist a group and create it otherwise
+[[ $(getent group $GID_) ]] || groupadd -g $GID_ $USERNAME_ &> /dev/null
 
-echo "$USERNAME_ ALL=(ALL) NOPASSWD:/usr/bin/apt-get" >> /etc/sudoers
-echo "$USERNAME_ ALL=(ALL) NOPASSWD:/usr/bin/apt" >> /etc/sudoers
-echo "$USERNAME_ ALL=(ALL) NOPASSWD:/usr/bin/dpkg" >> /etc/sudoers
+# check if a user with id $UID_ already exists
+if [[ $(id -u $UID_ &> /dev/null) ]]; then
+  # unconditionally set the primary group of the user
+  usermod -g $GID_ $UID_
+  # rename the user if required
+  [[ "$USERNAME_" == "$(id -un $UID_)" ]] || usermod -l $USERNAME_ $(id -un $UID_)
+else
+  # otherwise create the user
+  [[ -e /home/$USERNAME_ ]] && export m_= || export m_=m
+  useradd -${m_}d /home/$USERNAME_ -g $GID_ -s /bin/bash -u $UID_ $USERNAME_
+fi
 
-export LANG=en_US.UTF-8
-echo "127.0.0.1 ${HOSTNAME}" >> /etc/hosts
+# give the user some sudo capabilities
+if [[ $UID_ != 0 ]] ; then
```

Comment:
What's the benefit of guarding these?

Reply:
I just found it unnatural to add root to the sudoers file, objections?

Reply:
No, just found it weird.